core/extraction/towns_fund.py

```diff
@@ -547,6 +553,38 @@ def extract_funding_data(df_input: pd.DataFrame, project_lookup: dict) -> pd.Dat
 
     df_funding = convert_financial_halves(df_funding, "Reporting Period")
     df_funding.reset_index(drop=True, inplace=True)
+
+    # drop always unused cells TF funding before 2020
+    unused_mask = df_funding.loc[(df_funding["Secured"].isna()) & (df_funding["Start_Date"].isna())]
+    df_funding.drop(unused_mask.index, inplace=True)
+
+    if list(project_lookup.values())[0].split("-")[0] == "HS":
+        unused_fhsf_mask = df_funding.loc[
+            # drop unused FHSF Questions
+            (
+                df_funding["Funding Source Name"].isin(
+                    [
+                        "Town Deals 5% CDEL Pre-Payment",
+                        "Towns Fund RDEL Payment which is being utilised on TF project related activity",
+                        "How much of your RDEL forecast is contractually committed?",
+                    ]
+                )
+            )
+            |
+            # drop unused FHSF cells
+            (df_funding["Secured"].isna()) & (df_funding["Start_Date"] > end_of_reporting_period + timedelta(days=1))
```

Comment:
Looking at the form, this should only happen for the "mandatory" funding sources, not the custom ones. Need to adda bit of logic here, ie just remove these dates if the "funding source name" is one of the remaining 2 valid mandatory funding source names. Does that make sense?